h2/src/main/org/h2/engine/Session.java

```diff
@@ -703,6 +693,7 @@ public void commit(boolean ddl) {
             // (create/drop table and so on)
             database.commit(this);
         }
```

Comment:
I think you're going to need to copy part of the above "if" block into analyzeTables() to make this work properly, we will have a new open transaction and the META table will need committing